### PR TITLE
EIP-2315: updated testcases + clarifications

### PR DESCRIPTION
**Pull request by holiman**
This PR contains the updated testcases with the recent changes. Also some clarifications and I re-added cost of `BEGINSUB`. All opcodes should have a defined cost, IMO.

### CODE REVIEW

File: EIPS/eip-2315.md [hunk 1]
```diff
@@ -149,11 +147,23 @@ Bytecode: `0x600556b2b75b6003b3`
...
+
+In this example, the code 'walks' into a subroutine, which is not allowed, and causes an error
+
+|  Pc   |      Op     | Cost |   Stack   |   RStack  |
+|-------|-------------|------|-----------|-----------|
+|    0  |   BEGINSUB  |    1 |        [] |        [] |
+
+
+```
+Error: at pc=0, op=BEGINSUB: invalid subroutine entry
```
Comment (matkt) [inline EIPS/eip-2315.md:165]: Do we want an OOG message or another like `invalid subroutine entry` ? 
If we want to use this new message maybe we need to update this part https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2315.md#beginsub ?

**Comment by holiman** [comment]
The actual error message is not part of consensus

**Comment by matkt** [comment]
> The actual error message is not part of consensus

yes indeed it's just to have a consistency :)

**Comment by holiman** [comment]
Well, I think saying "it goes OOG", means that it will "consensus-wise" behave as an OOG, which is a well-understood failure mode at this time. Whether the human-readable strings that a client spits out says that or something else is not something that should be defined in the EIP, imo.

If we want more consistency, it would be more suited for another eip (not `core`), to standardize stuff that are non-consensus-critical. 

But I guess it's un-intuitive that the EIP says OOG error but the EIP-examples says something else. SO may be I should clarify in the EIP that the testcases/examples are based on geth and the message itself is an implementation detail.

**Pull request merged**